timelock-impl/src/main/java/com/palantir/atlasdb/timelock/paxos/ClientAwarePaxosLearner.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.paxos;
+
+import java.util.Collection;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import com.palantir.common.annotation.Inclusive;
+import com.palantir.paxos.PaxosValue;
+
+@Path("/" + PaxosTimeLockConstants.INTERNAL_NAMESPACE
+        + "/" + PaxosTimeLockConstants.CLIENT_PAXOS_NAMESPACE
+        + "/{client}"
+        + "/learner")
+public interface ClientAwarePaxosLearner {
+
+    /**
+     * Learn given value for the seq-th round.
+     *
+     * @param seq round in question
+     * @param val value learned for that round
+     */
+    @POST
+    @Path("learn/{seq:.+}")
```

Comment:
should `seq` be `[0-9]+` (or `-?[0-9]+`??)

Reply:
is fine, matches existing behaviour, and maybe getting an (I guess?) `NumberFormatException` is better than not-found.